src/main/java/com/direwolf20/buildinggadgets/common/construction/modes/CircleMode.java

```diff
@@ -0,0 +1,71 @@
+package com.direwolf20.buildinggadgets.common.construction.modes;
+
+import com.direwolf20.buildinggadgets.common.construction.ModeUseContext;
+import com.direwolf20.buildinggadgets.common.construction.XYZ;
+import com.direwolf20.buildinggadgets.common.schema.BoundingBox;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.util.Direction;
+import net.minecraft.util.math.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CircleMode extends Mode {
+    private final boolean filled;
+    public CircleMode(boolean isExchanging, boolean filled) {
+        super(filled ? "circle" : "hollow_circle", isExchanging);
+        this.filled = filled;
+    }
+
+    @Override
+    List<BlockPos> collect(ModeUseContext context, PlayerEntity player, BlockPos start) {
+        List<BlockPos> coordinates = new ArrayList<>();
+        int radius = context.getRange();
+        Direction direction = context.getHitSide();
+
+        int x = radius;
+        int y = 0;
+        int correction = 1 - x;
+
+        while(x >= y) {
+            if (this.filled) {
+                for (int i = -x; i <= x; i++) {
+                    coordinates.add(this.getBlockPosition(start, i, y, direction));
+                    if (y != 0) coordinates.add(this.getBlockPosition(start, i, -y, direction));
+                }
+                if (x != y) {
+                    for (int i = -y; i <= y; i++) {
+                        coordinates.add(this.getBlockPosition(start, i, x, direction));
+                        if (x != 0) coordinates.add(this.getBlockPosition(start, i, -x, direction));
+                    }
+                }
+            } else {
+                coordinates.add(this.getBlockPosition(start, x, y, direction));
```

Comment:
Try a turnary here instead of the if block blow, should still be simple enough to read. if it looks messy then ignore this :+1:

Reply:
I don't see what you mean exactly

Reply:
We discussed this I think, It's all good, thius is no longer required :+1: